src/IIIFAuth2/IIIFAuth2.API/Infrastructure/Auth/AuthAspectManager.cs

```diff
@@ -89,34 +91,29 @@ public void RemoveCookieFromResponse(int customerId)
     {
         const string bearerTokenScheme = "bearer";
 
-        var requestHeaders = GetContext().Request.Headers;
+        var requestHeaders = httpContextAccessor.SafeHttpContext().Request.Headers;
         return AuthenticationHeaderValue.TryParse(requestHeaders.Authorization, out var parsed) &&
                parsed.Scheme.Equals(bearerTokenScheme, StringComparison.InvariantCultureIgnoreCase)
             ? parsed.Parameter
             : null;
     }
     
-    private void HandleCookieDomain(int customer, DomainCookieHandler domainCookieHandler)
+    private async Task HandleCookieDomain(int customerId, DomainCookieHandler domainCookieHandler)
     {
-        var httpContext = GetContext();
-        var domains = GetCookieDomainList(httpContext);
+        var httpContext = httpContextAccessor.SafeHttpContext();
+        var domains = await GetCookieDomainList(customerId, httpContext);
 
-        var cookieId = GetAuthCookieKey(authSettings.CookieNameFormat, customer);
+        var cookieId = GetAuthCookieKey(authSettings.CookieNameFormat, customerId);
 
         foreach (var domain in domains)
         {
             domainCookieHandler(httpContext, cookieId, domain);
         }
     }
 
-    private HttpContext GetContext() =>
-        httpContextAccessor.HttpContext.ThrowIfNull(nameof(httpContextAccessor.HttpContext));
-
-    private IEnumerable<string> GetCookieDomainList(HttpContext httpContext)
+    private async Task<IEnumerable<string>> GetCookieDomainList(int customerId, HttpContext httpContext)
     {
-        var domains = authSettings.CookieDomains;
-        return authSettings.UseCurrentDomainForCookie
-            ? domains.Union(httpContext.Request.Host.Host.AsList())
-            : domains;
+        var customDomains = await customerDomainProvider.GetCustomerCookieDomains(customerId);
+        return customDomains.Union(httpContext.Request.Host.Host.AsList()).Distinct();
```

Comment:
If this is only ever going to return distinct values, would using a `HashSet` be better?